### PR TITLE
LPA-3440 Public Front Auto-Scaling

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
On this PR for LPA-3440, I’ve set up ecs autoscaling for the public front end, tracking cpu and memory usage. I’ve also brought the task size down to an eighth.

There’s a cloudwatch metric alarm tracking the number of ecs tasks deployed.

I've added a simple load test using locust, and a bit of faker to generate random people each pass of the user journey. It’s not an intention to run load tests on the pipeline or anything regularly, but they’re there if someone wants to check it i future.

Adds a PR template (included after this PR was started), and terraform pre-commit checks

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Fixes [LPA-####](https://opgtransform.atlassian.net/browse/LPA-####)
+
+## Approach
+
+_Explain how your code addresses the purpose of the change_
+
+## Learning
+
+_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about DigiDeps_
```
Comment (gregtyler) [inline docs/pull_request_template.md]:
```suggestion
_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_
```
Much as I love cross-team working, y'all don't want our baggage

Comment (andrewpearce-digital) [reply docs/pull_request_template.md]: Thanks

File: docs/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## Purpose
+_Briefly describe the purpose of the change, and/or link to the JIRA ticket for context_
+
+Fixes [LPA-####](https://opgtransform.atlassian.net/browse/LPA-####)
```
Comment (andrewpearce-digital) [inline docs/pull_request_template.md]:
```suggestion
Fixes LPA-####
```

File: tests/load_testing/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+We use [Faker](https://github.com/joke2k/faker) and some random number generation to create unique claims for each time locust runs a complete journey of the claim a refund public front end.
+
+The load test script will refuse to run against protected environments, checking the target url.
+
+### Install
+
+Install python dependencies using pip
+
+``` bash
+pip install -r requirements.txt
```
Comment (mattmachell) [inline tests/load_testing/README.md]: Is it worth containerising this? Would ease python environment managing.

File: docs/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+## Learning
+
+_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_
+
+## Checklist
+
+* [ ] I have performed a self-review of my own code
+* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
+* [ ] I have added tests to prove my work
+* [ ] The product team have tested these changes
```
Comment (mattmachell) [inline docs/pull_request_template.md]: Phrasing here might need a tweak? It has been demoed to product manager?

Comment (andrewpearce-digital) [reply docs/pull_request_template.md]: We're yet to agree as a team what this template should include, I'm getting a headstart. There's things we can agree to and reference here about our agreed best practice for testing, where docs should go, and how we get acceptance.

File: tests/load_testing/README.md (hunk 3 above)
Comment (andrewpearce-digital) [reply tests/load_testing/README.md]: A container for this has been hard to get working. 
Using available images, I can't meet the dependencies for locust, realbrowserlocust and firefox.
The solution might be to make a new build process that does it all.
I'll raise this with the team at next planning and we can talk about the value of being able to run this in a container.

**Pull request merged**